corehq/apps/translations/app_translations/upload_module.py

```diff
@@ -115,12 +115,10 @@ def _update_case_search_labels(self, rows):
         properties = self.module.search_config.properties
         displays = [row for row in self.condensed_rows if row['list_or_detail'] == 'case_search_display']
         hints = [row for row in self.condensed_rows if row['list_or_detail'] == 'case_search_hint']
-        if len(displays) != len(hints) or len(displays) != len(properties):
-
+        if len(displays) != len(properties):
             message = _(
-                'Expected {expected_count} case_search_display and case_search_hint '
-                'properties in  menu {index}, found {actual_label_count} for case_search_display and '
-                '{actual_hint_count} for case_search_hint'
+                'Expected {expected_count} case_search_display '
+                'properties in  menu {index}, found {actual_label_count} for case_search_display. '
```

Comment:
nit: Is this message "No Case Search config properties for menu {index} were updated." still accurate? The case config property's hint text could potentially be updated now right?

Also same comment for the other messages
```
message = _('A display row for menu {index} has an unexpected case search property "{field}". '
                                'Case properties must appear in the same order as they do in the bulk '
                                'app translation download. No translations updated for this row.').format(...
```

Reply:
nope this message is still accurate - it won't get into updating either the displays or hints if the number of displays isn't correct (everything happens in that else block)

Reply:
oh i see, I missed that the `if len(hints) == len(properties):` block was nested inside the `else`. In that case, is there a reason to continue making the hints text updates contingent on the ` if len(displays) != len(properties):`? As in, would it make sense to do this 
```
if len(displays) != len(properties):
...
else:
...
if len(hints) == len(properties):
...
elif...
```

Reply:
hmm I guess not? I don't know what the rationale was in the beginning having nothing update if there wasn't the proper amount of displays/hint rows. I can ask Sravan tomorrow but it's likely I can also have the display update partially too + change the error message to whatever is not the red banner with something like "we updated some rows but noticed you were missing some". Like I said in another comment, users aren't likely to hit this unless they manually edit the excel file but probably still good to have.